### PR TITLE
Use `ptr::{null, null_mut}` instead of `0 as *{const, mut}`

### PR DESCRIPTION
**Pull request by tbu-**

### CODE REVIEW

**Comment by aturon** [comment]
Thanks!

@bors: r+

File: src/libstd/sys/unix/time.rs [hunk 1]
```diff
@@ -169,7 +169,7 @@ mod inner {
                 tv_usec: 0,
             };
             cvt(unsafe {
-                libc::gettimeofday(&mut s, 0 as *mut _)
+                libc::gettimeofday(&mut s, ptr::null_mut())
```
Comment (apasel422) [inline src/libstd/sys/unix/time.rs]: Seems that this file is missing an import of `ptr`.

Comment (tbu-) [inline src/libstd/sys/unix/time.rs]: Should be fixed.

**Comment by apasel422** [comment]
Travis output:
```
src/libstd/sys/unix/time.rs:13:5: 13:8 error: unused import
src/libstd/sys/unix/time.rs:13 use ptr;
                                   ^~~
src/libstd/lib.rs:289:31: 289:39 note: lint level defined here
src/libstd/lib.rs:289 #![cfg_attr(not(stage0), deny(warnings))]
```

**Comment by apasel422** [comment]
Travis output:
```
src/libtest/lib.rs:71:5: 71:13 error: unused import
src/libtest/lib.rs:71 use std::ptr;
                          ^~~~~~~~
src/libtest/lib.rs:34:31: 34:39 note: lint level defined here
src/libtest/lib.rs:34 #![cfg_attr(not(stage0), deny(warnings))]
```

**Comment by tbu-** [comment]
Travis successfully compiled the code, so it should be good now.

**Comment by Manishearth** [comment]
https://buildbot.rust-lang.org/builders/auto-win-gnu-64-opt/builds/4820

**Comment by tbu-** [comment]
Fixed.

**Comment by tbu-** [comment]
Rebased.

**Pull request merged**